l10n_eu_product_adr_local/views/adr_goods_views.xml

```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+
+    <record id="adr_goods_view_form" model="ir.ui.view">
+        <field name="name">adr.goods (in l10n_eu_product_adr_local) </field>
+        <field name="model">adr.goods</field>
+        <field name="inherit_id" ref="l10n_eu_product_adr.adr_goods_view_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='un_number']" position="before">
+                <field name="un_local_code"/>
+            </xpath>
+            <xpath expr="//field[@name='packing_instruction_ids']" position="before">
+                <field name="packaging_group_id" options="{'no_open': True, 'no_create': True}"/>
+            </xpath>
+        </field>
+    </record>
+
+    <record id="adr_goods_view_tree" model="ir.ui.view">
+        <field name="name">adr.goods (in l10n_eu_product_adr_local) </field>
+        <field name="model">adr.goods</field>
+        <field name="inherit_id" ref="l10n_eu_product_adr.adr_goods_view_tree"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='un_number']" position="before">
+                <field name="un_local_code"/>
+            </xpath>
```

Comment:
For `tree` view inheritance you can try a simpler definition:
```suggestion
            <field name="un_number" position="before">
                <field name="un_local_code"/>
            </field>
```

`xpath` expressions are always a good option, but strictly required for `form` inheritances, because they are usually more complex and require more precise paths.